homeassistant/components/advantage_air/climate.py

```diff
@@ -0,0 +1,278 @@
+    "auto": FAN_AUTO,
+    "low": FAN_LOW,
+    "medium": FAN_MEDIUM,
+    "high": FAN_HIGH,
+}
+HASS_FAN_MODES = {v: k for k, v in ADVANTAGE_AIR_FAN_MODES.items()}
+FAN_SPEEDS = {FAN_LOW: 30, FAN_MEDIUM: 60, FAN_HIGH: 100}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up AdvantageAir climate platform."""
+
+    instance = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities = []
+    for ac_key in instance["coordinator"].data["aircons"]:
+        entities.append(AdvantageAirAC(instance, ac_key))
+        for zone_key in instance["coordinator"].data["aircons"][ac_key]["zones"]:
+            # Only add zone climate control when zone is in temperature control
+            if (
+                instance["coordinator"].data["aircons"][ac_key]["zones"][zone_key][
+                    "type"
+                ]
+                != 0
+            ):
+                entities.append(AdvantageAirZone(instance, ac_key, zone_key))
+    async_add_entities(entities)
+    return True
+
+
+class AdvantageAirClimateEntity(CoordinatorEntity, ClimateEntity):
+    """AdvantageAir Climate class."""
+
+    def __init__(self, instance):
+        """Initialize the base Advantage Air climate entity."""
+        super().__init__(instance["coordinator"])
+        self.async_change = instance["async_change"]
+        self.device = instance["device"]
+
+    @property
+    def temperature_unit(self):
+        """Return the temperature unit."""
+        return TEMP_CELSIUS
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported temperature step."""
+        return PRECISION_WHOLE
+
+    @property
+    def max_temp(self):
+        """Return the maximum supported temperature."""
+        return 32
+
+    @property
+    def min_temp(self):
+        """Return the minimum supported temperature."""
+        return 16
+
+    @property
+    def device_info(self):
+        """Return parent device information."""
+        return self.device
+
+
+class AdvantageAirAC(AdvantageAirClimateEntity):
+    """AdvantageAir AC unit."""
+
+    def __init__(self, instance, ac_key):
+        """Initialize the Advantage Air AC climate entity."""
+        super().__init__(instance)
+        self.ac_key = ac_key
+
+    @property
+    def name(self):
+        """Return the name."""
+        return self.coordinator.data["aircons"][self.ac_key]["info"]["name"]
+
+    @property
+    def unique_id(self):
+        """Return a unique id."""
+        return f'{self.coordinator.data["system"]["rid"]}-{self.ac_key}'
+
+    @property
+    def target_temperature(self):
+        """Return the current target temperature."""
+        return self.coordinator.data["aircons"][self.ac_key]["info"]["setTemp"]
+
+    @property
+    def hvac_mode(self):
+        """Return the current HVAC modes."""
+        if self.coordinator.data["aircons"][self.ac_key]["info"]["state"] == STATE_ON:
```

Comment:
Don't use home assistant constants to compare with the device api values. We shouldn't couple the two APIs like that. Define new constants in this module to use for the device api.

Reply:
Replaced the Home Assistant constants with new constants in const.py as they are used in other platforms (not currently in this PR).